ReactWindows/ReactNative/Views/Text/ReactTextViewManager.cs

```diff
@@ -185,5 +187,203 @@ protected override RichTextBlock CreateViewInstance(ThemedReactContext reactCont
+                                return true;
+                            }
+                        }
+                        break;
+                    // Also covers Bold, Italic, Underline.
+                    case Span span:
+                        if (ProcessSelectedInlines(span.Inlines, selectedText, accumulate, selectionStart, selectionEnd))
+                        {
+                            return true;
+                        }
+                        break;
+                    case Run run:
+                        if (accumulate)
+                        {
+                            int rangeStart = 0;
+                            // Check for selection start in this run.
+                            if (object.ReferenceEquals(inline, selectionStart.Parent))
+                            {
+                                rangeStart = selectionStart.Offset - inline.ContentStart.Offset;
+                            }
+                            // Check for selection end in this run.
+                            if (object.ReferenceEquals(inline, selectionEnd.Parent))
+                            {
+                                int rangeEnd = selectionEnd.Offset - inline.ContentStart.Offset;
+                                selectedText.Append(run.Text.Substring(rangeStart, rangeEnd - rangeStart));
+                            }
+                            else
+                            {
+                                selectedText.Append(run.Text.Substring(rangeStart));
+                            }
+                        }
+                        break;
+                    case LineBreak lineBreak:
+                        if (accumulate)
+                        {
+                            selectedText.AppendLine();
+                        }
+                        break;
+                    case InlineUIContainer inlineUIContainer:
+                        if (accumulate)
+                        {
+                            // Since InlineUIContainer can contain any UIElement as a child, we'll make some assumptions in order to simplify processing.
+                            // 1. Entire InlineUIContainer content is considered selected.
+                            // 2. If the child is a TextBlock, its Text property returns the entire content properly (since it can't contain InlineUIContainers).
+                            // 3. If the InlineUIContainer Child is a Panel, we process only TextBlock children (and not RichTextBlocks) so that we limit recursion.
+                            if (inlineUIContainer.Child is TextBlock textBlock)
+                            {
+                                selectedText.Append(textBlock.Text);
+                            }
+                            else if (inlineUIContainer.Child is Panel panel)
+                            {
+                                foreach (UIElement uiElement in panel.Children)
+                                {
```

Comment:
Seems like we should either support extracting text from an InlineUIContainer completely or not at all. I'm comfortable with "skipping" this element for now. This approach will not cover even one level of indirection where you have text embedded in more than one panel.

Reply:
@rozele When you say extracting text completely do you mean to continue recursion further if InlineUIContainer contains other RichTextBlocks. Since it can practically contain any kind of control (in theory) but practically it is usually some image/emoticon and text for description I am unsure how deep this should go and what kind of controls should be considered?

---
In reply to: [215226616](https://github.com/Microsoft/react-native-windows/pull/1951#discussion_r215226616) [](ancestors = 215226616)

Reply:
I guess if we just document the conventions we create, and include a React.js sample in the comments or markdown (somewhere) for the kinds of components that would be extracted as text (and not be extracted as well), it's okay to do this kind of shallow grab for text.

Reply:
@rozele I added full recursion for an embedded RichTextBlock and also a description of the limitations of the limitations in the proc description.

---
In reply to: [215359796](https://github.com/Microsoft/react-native-windows/pull/1951#discussion_r215359796) [](ancestors = 215359796,215226616)

Reply:
Done, added clarification along with extra information react.js sample markup in the proc description

---
In reply to: [215467085](https://github.com/Microsoft/react-native-windows/pull/1951#discussion_r215467085) [](ancestors = 215467085)